docs/writing-rules/data-flow/taint-mode.md

```diff
@@ -306,3 +630,104 @@ Semgrep can also treat metavariables with the same name as the _same_ metavariab
+
+<iframe src="https://semgrep.dev/embed/editor?snippet=PeBXv" border="0" frameBorder="0" width="100%" height="432"></iframe>
+
+Using the CLI option `--pro-intrafile`, Semgrep will perform inter-procedural (across functions) _intra_-file (within one file) analysis. That is, it will track taint across functions, but it will not cross file boundaries. This is supported for essentially every language, and performance is very close to that of intra-procedural taint analysis.
+
+Using the CLI option `--pro`, Semgrep will perform inter-procedural (across functions) as well as *inter*-file (across files) analysis. Inter-file analysis is only supported for [a subset of languages](/supported-languages/#semgrep-code-language-support). For a rule to run inter-file it also needs to set `interfile: true`:
+
+```yaml
+options:
+  interfile: true
+```
+
+**Memory requirements for inter-file analysis:**
+While inter-file analysis is more powerful, be aware that it also demands more memory resources. We advise a minimum of 4 GB of memory per core, but _recommend 8 GB per core or more_. The amount of memory needed depends on the codebase and on the number of inter-file rules being run.
+
+## How sensitive is taint-mode ?
+
+### Field sensitivity
+
+The taint engine provides basic field sensitivity support, it can:
+
+- Track that `x.a.b` is tainted, but `x` or `x.a` is  **not** tainted. If `x.a.b` is tainted, any extension of `x.a.b` (such as `x.a.b.c`) is considered tainted by default.
+- Track that `x.a` is tainted, but remember that `x.a.b` has been sanitized. Thus the engine records that `x.a.b` is **not** tainted, but `x.a` or `x.a.c` are still tainted.
+
+:::note
+The taint engine does track taint **per variable** and not **per object in memory**. The taint engine does not track aliasing at present.
+:::
+
+<iframe src="https://semgrep.dev/embed/editor?snippet=5rvkj" border="0" frameBorder="0" width="100%" height="432"></iframe>
+
+### Index sensitivity (Pro)
+
+**Index sensitivity is a Semgrep Pro feature.**
+
+Semgrep Pro has basic index sensitivity support:
+- Only for accesses using the built-in `a[E]` syntax.
+- Works for _statically constant_ indexes that may be either integers (e.g. `a[42]`) or strings (e.g. `a["foo"]`).
+- If an arbitrary index `a[i]` is sanitized, then every index becomes clean of taint.
+
+<iframe src="https://semgrep.dev/embed/editor?snippet=GdoK6" border="0" frameBorder="0" width="100%" height="432"></iframe>
+
+## Taint labels (Pro) 🧪
+
+Taint labels increase the expressiveness of taint analysis by allowing you to specify and track different kinds of tainted data in one rule using labels. This functionality has various uses, for example, when data becomes dangerous in several steps that are hard to specify through single pair of source and sink.
+
+<iframe class="yt_embed" width="100%" height="432px" src="https://www.youtube.com/embed/lAbJdzMUR4k" frameborder="0" allowfullscreen></iframe>
```

Comment:
kyle
 [2 hours ago](https://semgrepinc.slack.com/archives/C02JT6YA5S7/p1711423039363669)
Yello!  I think there is an error on this page:
https://semgrep.dev/docs/writing-rules/experiments/taint-labels/
I believe the test code should look like:
def bad(user_input):
    x = user_input
    # ok: taint-labels (no evil!)
    sink(x)
rather than the current code, which shows sink(y) :slightly_smiling_face:

observation from @jkylekelly

Reply:
Is this related to Pieter's video?

Reply:
Found the right line. Resolving this one.